google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/AppEngineApplicationCreateDialog.java

```diff
@@ -0,0 +1,206 @@
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.ui.JBColor;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.Component;
+import java.awt.event.ItemEvent;
+import java.io.IOException;
+import java.util.List;
+
+import javax.swing.JComboBox;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+import javax.swing.JTextPane;
+
+/**
+ * Dialog that allows creation of an App Engine Application
+ */
+public class AppEngineApplicationCreateDialog extends DialogWrapper {
+
+  private final static String LOCATIONS_DOCUMENTATION_URL
+      = "https://cloud.google.com/docs/geography-and-regions";
+  private final static String FLEX_DOCUMENTATION_URL
+      = "https://cloud.google.com/appengine/docs/flexible/";
+  private final static String STANDARD_DOCUMENTATION_URL
+      = "https://cloud.google.com/appengine/docs/about-the-standard-environment";
+  private static final String HTML_OPEN_TAG = "<html><font face='sans' size='-1'>";
+  private static final String HTML_CLOSE_TAG = "</font></html>";
+
+  private JPanel panel;
+  private JTextPane instructionsTextPane;
+  private JComboBox<AppEngineLocationSelectorItem> regionComboBox;
+  private JTextPane statusPane;
+  private JTextPane regionDetailPane;
+
+  private final Credential userCredential;
+  private final String gcpProjectId;
+
+  public AppEngineApplicationCreateDialog(@NotNull Component parent, @NotNull String gcpProjectId,
+      @NotNull Credential userCredential) {
+    super(parent, false);
+
+    this.gcpProjectId = gcpProjectId;
+    this.userCredential = userCredential;
+
+    init();
+    setTitle(GctBundle.message("appengine.application.region.select"));
+    refreshLocationsSelector();
+
+    regionComboBox.addItemListener((event) -> {
+      if (event.getStateChange() == ItemEvent.SELECTED) {
+        updateLocationDetailMessage();
+      }
+    });
+
+    regionDetailPane.addHyperlinkListener(new BrowserOpeningHyperLinkListener());
+    instructionsTextPane.addHyperlinkListener(new BrowserOpeningHyperLinkListener());
+    instructionsTextPane.setText(HTML_OPEN_TAG
+        + GctBundle.message("appengine.application.create.instructions") + "<p>"
+        + GctBundle.message("appengine.application.create.documentation",
+        "<a href=\"" + LOCATIONS_DOCUMENTATION_URL + "\">", "</a>") + "</p>" + HTML_CLOSE_TAG);
+  }
+
+  @Override
+  protected void doOKAction() {
+    final Location selectedLocation
+        = ((AppEngineLocationSelectorItem) regionComboBox.getSelectedItem()).getLocation();
+
+    // show loading state
+    setOKActionEnabled(false);
+
+    try {
+      UsageTrackerProvider.getInstance()
+          .trackEvent(GctTracking.APP_ENGINE_APPLICATION_CREATE)
```

Comment:
Good point, though adding metrics doesn't cost us anything, so I was thinking it would just be nice to include anyway. 

Thinking about it now, there are some situations where this could be valuable - since this is a long-running operation, it could be good to track requests made separately from completions, e.g. if the user cancels the request or kills the application while it's running.